core/src/main/java/ai/timefold/solver/core/config/util/ConfigUtils.java

```diff
@@ -87,8 +91,9 @@ public static <T> T newInstance(Supplier<String> ownerDescriptor, String propert
         }
     }
 
-    public static void applyCustomProperties(Object bean, String beanClassPropertyName,
-            Map<String, String> customProperties, String customPropertiesPropertyName) {
+    // TODO: param customPropertiesPropertyName
+    public static void applyCustomProperties(@NonNull Object bean, @NonNull String beanClassPropertyName,
+            @Nullable Map<@NonNull String, @NonNull String> customProperties, String customPropertiesPropertyName) {
```

Comment:
@triceo `customPropertiesPropertyName` is also only used for exception messages, is this `@Nullable`?